src/ck-core/charm++.h

```diff
@@ -234,10 +234,13 @@ class Chare {
 #ifndef CMK_CHARE_USE_PTR
     int chareIdx;                  // index in the chare obj table (chare_objs)
 #endif
+    int r_count;
+    bool atsync, atsync_notify;
     Chare(CkMigrateMessage *m);
     Chare();
     virtual ~Chare(); //<- needed for *any* child to have a virtual destructor
-
+    virtual void ResumeFromSync(void);
```

Comment:
Is there a reason this is moved up to Chare? AFAIK the only things that relate to AtSync are CkMigratables.

Reply:
ResumeFromSync is also called on CkLocMgr for its dummyResumeFromSync() which I think is used to call Registering and DoneRegistering at every AtSync. Since both CkMigratable and CkLocMgr have ResumeFromSync method, I moved it to Chare.